### PR TITLE
Introduce log level and format config

### PR DESCRIPTION
**Pull request by tareqmamari**
This commit introduce support of configuring log level and log format.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -257,12 +263,30 @@ func SetupDaemonConfig(logger *logrus.Logger, configFile string) (DaemonConfig,
...
+
 	setter.SetDefault(&DebugEnabled, getEnvBool(log, "GUBER_DEBUG"))
 	if DebugEnabled {
 		logger.SetLevel(logrus.DebugLevel)
 		log.Debug("Debug enabled")
 	}
 
+	setter.SetDefault(&logConfig.LogLevel, uint32(getEnvInteger(log, "GUBER_LOG_LEVEL")))
+	if !DebugEnabled {
+		logger.SetLevel(logrus.Level(logConfig.LogLevel))
```
Comment (Baliedge) [inline config.go:287]: Traditionally at Mailgun, we use log level strings.  That way, we can use constants like `logrus.DebugLevel` or "DEBUG" in config files.  For this, use data type `logrus.Level`.

File: config.go [hunk 2]
```diff
@@ -257,12 +263,30 @@ func SetupDaemonConfig(logger *logrus.Logger, configFile string) (DaemonConfig,
...
+	}
+
 	setter.SetDefault(&DebugEnabled, getEnvBool(log, "GUBER_DEBUG"))
 	if DebugEnabled {
 		logger.SetLevel(logrus.DebugLevel)
 		log.Debug("Debug enabled")
 	}
 
+	setter.SetDefault(&logConfig.LogLevel, uint32(getEnvInteger(log, "GUBER_LOG_LEVEL")))
+	if !DebugEnabled {
```
Comment (Baliedge) [inline config.go:286]: This logic is tied to line 280.  It would make sense this be together in the same `if`/`else` condition.

File: config.go [hunk 3]
```diff
@@ -164,6 +164,11 @@ func (p PeerInfo) HashKey() string {
 
 type UpdateFunc func([]PeerInfo)
 
+type LogConfig struct {
```
Comment (Baliedge) [inline config.go:167]: It doesn't look like we need a struct to define these fields.  Can they just be variables in `SetupDaemonConfig()`?

**Review by Baliedge — CHANGES_REQUESTED** [review]

**Comment by Baliedge** [comment]
@tareqmamari Thanks for the contribution!  I left just a few minor comments to address before I merge.

File: config.go (hunk 1 above)
Comment (tareqmamari) [reply config.go]: @Baliedge I just refactored that, could you please have a look again?

**Comment by tareqmamari** [comment]
@Baliedge Thank you for the review. I just addressed your comments, please let me know if anything else is needed.

File: config.go (hunk 1 above)
Comment (Baliedge) [reply config.go]: I was thinking more along the lines of:
```go
	setter.SetDefault(&DebugEnabled, getEnvBool(log, "GUBER_DEBUG"))
	setter.SetDefault(&logLevel, os.Getenv("GUBER_LOG_LEVEL"))
	if DebugEnabled {
		logger.SetLevel(logrus.DebugLevel)
		log.Debug("Debug enabled")
	} else if logLevel != "" {
		logrusLogLevel, err := logrus.ParseLevel(logLevel)
		if err != nil {
			return conf, errors.Wrap(err, "invalid log level")
		}

		logger.SetLevel(logrusLogLevel)
	}
```

File: config.go (hunk 2 above)
Comment (Baliedge) [reply config.go]: See code example in previous comment.

Comment (tareqmamari) [reply config.go]: Oops, I just missed this comment @Baliedge, let me update it.

File: config.go (hunk 1 above)
Comment (tareqmamari) [reply config.go]: @Baliedge exactly, I would have done it the same, it should be Ok now.

**Review by Baliedge — APPROVED** [review]
LGTM

**Pull request merged**